test/flows/student_finance_calculator_flow_test.rb

```diff
@@ -0,0 +1,362 @@
+require "test_helper"
+require "support/flow_test_helper"
+
+class StudentFinanceCalculatorTest < ActiveSupport::TestCase
+  include FlowTestHelper
+
+  setup { testing_flow StudentFinanceCalculatorFlow }
+
+  should "render a start page" do
+    assert_rendered_start_page
+  end
+
+  context "question: when_does_your_course_start?" do
+    setup { testing_node :when_does_your_course_start? }
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of what_type_of_student_are_you? for any response" do
+        assert_next_node :what_type_of_student_are_you?, for_response: "2020-2021"
+      end
+    end
+  end
+
+  context "question: what_type_of_student_are_you?," do
+    setup do
+      testing_node :what_type_of_student_are_you?
+      add_responses when_does_your_course_start?: "2020-2021"
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of how_much_are_your_tuition_fees_per_year? for any response" do
+        assert_next_node :how_much_are_your_tuition_fees_per_year?, for_response: "uk-full-time"
+      end
+    end
+  end
+
+  context "UK full time students" do
+    context "question: how_much_are_your_tuition_fees_per_year?," do
+      setup do
+        testing_node :how_much_are_your_tuition_fees_per_year?
+        add_responses when_does_your_course_start?: "2020-2021",
+                      what_type_of_student_are_you?: "uk-full-time"
+      end
+
+      should "render the question" do
+        assert_rendered_question
+      end
+
+      context "validation" do
+        should "be invalid if a fee over 9250 is entered for full time students" do
+          assert_invalid_response "9251"
+        end
+      end
+
+      context "next_node" do
+        should "have a next node of where_will_you_live_while_studying?" do
```

Comment:
We should have tests here for the next steps you can reach from this node and use `add_responses` to replace the previous response. There's an example of that in [inherit_someone_dies_without_will_flow_test.rb](https://github.com/alphagov/smart-answers/blob/65c061b5fb62ba0d81267bd269b492d4789cdae0/test/flows/inherits_someone_dies_without_will_flow_test.rb#L46-L59).

So I think you'd have something like:

```
%w[uk-full-time uk-part-time].each do |uk_student_type|
  should "have a next node of where_will_you_live_while_studying? for a #{uk_student_type} student" do
    add_responses what_type_of_student_are_you?: uk_student_type
    assert_next_node :where_will_you_live_while_studying?, for_response: "100"
  end
end

%w[eu-full-time eu-part-time].each do |eu_student_type|
  should "have a next node of outcome_eu_students for a #{eu_student_type} student" do
    add_responses what_type_of_student_are_you?: eu_student_type
    assert_next_node :outcome_eu_students, for_response: "100"
  end
end
```